docker.sh

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+export CMAKE_VERSION="3.15.2"
+
+main() {
+  if [[ -z "${run_build_script}" ]] ; then
+    startDockerContainer || return 1
+  else
+    buildOsqueryToolchain || return 1
+  fi
+
+  return 0
+}
+
+startDockerContainer() {
+  if [[ -d "build" ]] ; then
+    echo "Reusing existing build folder"
+  else
+    echo "Creating new build folder"
+
+    mkdir build
+    if [[ $? != 0 ]] ; then
+      echo "Failed to create the build folder"
+      return 1
+    fi
+  fi
+
+  local container_name="osquery-toolchain-$(git rev-parse HEAD)"
+  docker rm "${container_name}" > /dev/null 2>&1
+
+  docker run --rm -e "run_build_script=1" -v "$(realpath build):/opt/osquery-toolchain" -v "$(pwd):/home/osquery/osquery-toolchain" --name "${container_name}" -it ubuntu /bin/bash -c '/home/osquery/osquery-toolchain/docker.sh'
```

Comment:
Where does this container get build?

Reply:
This is directly using the default ubuntu image; the script will then initialize the running container and destroy it automatically after the build procedure has finished